src/nupic/ntypes/ArrayBase.cpp

```diff
@@ -190,72 +183,40 @@ const void *ArrayBase::getBuffer() const {
   return nullptr;
 }
 
-SDR& ArrayBase::getSDR() {
+sdr::SDR& ArrayBase::getSDR() {
   NTA_CHECK(type_ == NTA_BasicType_SDR) << "Does not contain an SDR object";
   if (buffer_ == nullptr) {
     std::vector<UInt> zeroDim;
     zeroDim.push_back(0u);
     allocateBuffer(zeroDim);  // Create an empty SDR object.
   }
-  SDR& sdr = *((SDR *)buffer_.get());
+  sdr::SDR& sdr = *((sdr::SDR *)buffer_.get());
```

Comment:
nit, can you use static_cast ? (or other relevant c++ casting), it does better checks that the plain C-cast

Reply:
> nit, can you use static_cast ?

I just ever got into the habit of using static_cast.  I know what C casting does.  Not so sure about the C++ casting.  But I guess I should lean to use the new stuff.

Reply:
have a look, it's a nice c++ feature: https://stackoverflow.com/questions/332030/when-should-static-cast-dynamic-cast-const-cast-and-reinterpret-cast-be-used 

My layman's version: 
- it's possible to grep sourcecode for the cast
- no performance penalty
- does some checks
- 1st goto version: `static_cast<T>(val)`

Reply:
..I'd like to see these implemented for the PR